File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/GroupOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.posthog.PosthogHelper;
import java.util.Map;
import org.json.JSONException;

public class GroupOptions {

    @NonNull
    private String type;

    @NonNull
    private String key;

    @Nullable
    private Map<String, Object> groupProperties;

    public GroupOptions(@NonNull String type, @NonNull String key, @Nullable JSObject groupProperties) throws JSONException {
        this.type = type;
        this.key = key;
        this.groupProperties = PosthogHelper.createHashMapFromJSONObject(groupProperties);
    }

    @NonNull
    public String getType() {
        return type;
    }

    @NonNull
    public String getKey() {
        return key;
    }

    @Nullable
    public Map<String, Object> getGroupProperties() {
        return groupProperties;
    }
}

File: packages/posthog/ios/Plugin/Classes/Options/GroupOptions.swift
import Foundation
import Capacitor

@objc public class GroupOptions: NSObject {
    private var type: String
    private var key: String
    private var groupProperties: [String: Any]?

    init(type: String, key: String, groupProperties: [String: Any]?) {
        self.type = type
        self.key = key
        self.groupProperties = groupProperties
    }

    func getType() -> String {
        return type
    }

    func getKey() -> String {
        return key
    }

    func getGroupProperties() -> [String: Any]? {
        return groupProperties
    }
}
